tests/unit/test.watcher.js

```diff
@@ -38,18 +50,47 @@ describe('watcher', () => {
             .then(() => watcher);
         })
         .then((watcher) => {
-          return whenFilesChanged
-            .then(() => {
+          return Promise.race([
+            whenFilesChanged
+              .then(() => {
+                watcher.close();
+                // This delay seems to avoid stat errors from the watcher
+                // which can happen when the temp dir is deleted (presumably
+                // before watcher.close() has removed all listeners).
+                return new Promise((resolve) => {
+                  setTimeout(resolve, 2, onChange);
+                });
+              }),
+            // Time out if no files are changed
+            new Promise((resolve) => setTimeout(() => {
               watcher.close();
-              sinon.assert.calledOnce(onChange);
-              // This delay seems to avoid stat errors from the watcher
-              // which can happen when the temp dir is deleted (presumably
-              // before watcher.close() has removed all listeners).
-              return new Promise((resolve) => setTimeout(resolve, 2));
-            });
```

Comment:
The way this was written before, `whenFilesChanged` will not resolve unless WatchPack actually detects the changes. This leads to a problem with the new feature because with the new option, if the detected file isn't modified, this will never resolve, which fails the test.

I added a "timeout" to the promise using `Promise.race()` – which after experimenting with seems to correctly fail the assertions if I want them to. The only bad thing about this I guess is that one test will always take 500ms (the timeout in that second promise). I don't think that's too bad, but if there's a better way to do all of this I'd love to learn 😄